### ISSUE TITLE
Can't build, mode_t declaration

### ISSUE THREAD

**Issue by taoeffect** [fluent/fluent-bit#1011]
## Bug Report

**Describe the bug**

When I try to build 1.0.1 on Alpine 3.8.2 I get:

```
[  2%] Built target onigmo
[  4%] Built target luajit
[  6%] Built target backtrace
[  7%] Built target jsmn
[  8%] Built target sqlite3
[  9%] Built target co
[ 10%] Built target xxd-c
Scanning dependencies of target flb-plugin-proxy-go
[ 10%] Building C object src/proxy/go/CMakeFiles/flb-plugin-proxy-go.dir/go.c.o
[ 10%] Linking C static library ../../../library/libflb-plugin-proxy-go.a
[ 10%] Built target flb-plugin-proxy-go
[ 12%] Built target msgpackc-static
[ 12%] Built target cio-crc32
[ 13%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_stream.c.o
In file included from /tmp/fluent-bit-1.0.1/lib/chunkio/src/cio_stream.c:26:0:
/tmp/fluent-bit-1.0.1/lib/chunkio/include/chunkio/cio_os.h:24:36: error: unknown type name 'mode_t'
 int cio_os_mkpath(const char *dir, mode_t mode);
                                    ^~~~~~
/tmp/fluent-bit-1.0.1/lib/chunkio/src/cio_stream.c: In function 'check_stream_path':
/tmp/fluent-bit-1.0.1/lib/chunkio/src/cio_stream.c:56:15: warning: implicit declaration of function 'cio_os_mkpath' [-Wimplicit-function-declaration]
         ret = cio_os_mkpath(p, 0755);
               ^~~~~~~~~~~~~
make[2]: *** [lib/chunkio/src/CMakeFiles/chunkio-static.dir/build.make:167: lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_stream.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:321: lib/chunkio/src/CMakeFiles/chunkio-static.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

**To Reproduce**

```
apk add apk-tools ca-certificates wget curl file nano cmake build-base fts fts-dev \
  && cd /root/fluent-bit-${FLUENT_VERSION}/build \
  && cmake \
    -DFLB_DEBUG=Off \
    -DFLB_TRACE=Off \
    -DFLB_TLS=On \
    -DFLB_JEMALLOC=Off \
    -DFLB_BUFFERING=On \
    -DFLB_SHARED_LIB=Off \
    -DFLB_EXAMPLES=Off \
    -DFLB_WITHOUT_SHARED_LIB=On \
    -DFLB_WITHOUT_EXAMPLES=On \
    .. \
  && make
```

**Expected behavior**

No error.

**Your Environment**

* Version used: 1.0.1
* Configuration: Docker on Mac running Alpine 3.8.2
* Environment name and version (e.g. Kubernetes? What version?):
* Server type and version:
* Operating System and version:
* Filters and plugins:

**Comment by taoeffect** [comment]
As a hack, you can fix this by adding this before the cmake or make step:

```
sed -i 's/#define CIO_OS_H/#define CIO_OS_H\n#include <sys\/types.h>/' ../lib/chunkio/include/chunkio/cio_os.h
```

**Comment by fujimotos** [comment]
FWIW, I'm actively working on this part of code right now.

I'll look after this issue, and make sure it get fixed along the way.

**Comment by fujimotos** [comment]
FWIW, this issue is now resolved in the current master HEAD
(since 1a0b6cb78 has been merged).

Indeed I can confirm that fluent-bit is compilable on Alpine 3.7,
so I think this ticket can be closed.

### Screenshot (Alpine v3.7)

![2019-02-05-104657_718x402_scrot](https://user-images.githubusercontent.com/8974561/52248844-a3d18900-2933-11e9-8d60-f5f61c0216b6.png)

**Comment by taoeffect** [comment]
OK, the merge looks right to me so I'll close, looking forward to the next release!

**Comment by taoeffect** [comment]
I am re-opening because attempting to build 1.0.6 results in pretty much the same error, and I had to re-add my hack above to get it to build.

```
[ 13%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_file.c.o
/root/fluent-bit-1.0.6/lib/chunkio/src/cio_file.c: In function 'cio_file_hash_print':
/root/fluent-bit-1.0.6/lib/chunkio/src/cio_file.c:652:23: warning: format '%lu' expects argument of type 'long unsigned int', but argument 2 has type 'crc_t' {aka 'unsigned int'} [-Wformat=]
     printf("crc cur=%lu\n", cf->crc_cur);
                     ~~^     ~~~~~~~~~~~
                     %u
[ 13%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_chunk.c.o
[ 14%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_meta.c.o
[ 14%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_scan.c.o
[ 14%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_utils.c.o
[ 15%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_stream.c.o
In file included from /root/fluent-bit-1.0.6/lib/chunkio/src/cio_stream.c:26:
/root/fluent-bit-1.0.6/lib/chunkio/include/chunkio/cio_os.h:24:36: error: unknown type name 'mode_t'; did you mean 'gid_t'?
 int cio_os_mkpath(const char *dir, mode_t mode);
                                    ^~~~~~
                                    gid_t
/root/fluent-bit-1.0.6/lib/chunkio/src/cio_stream.c: In function 'check_stream_path':
/root/fluent-bit-1.0.6/lib/chunkio/src/cio_stream.c:56:15: warning: implicit declaration of function 'cio_os_mkpath'; did you mean 'cio_os_isdir'? [-Wimplicit-function-declaration]
         ret = cio_os_mkpath(p, 0755);
               ^~~~~~~~~~~~~
               cio_os_isdir
make[2]: *** [lib/chunkio/src/CMakeFiles/chunkio-static.dir/build.make:167: lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_stream.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:326: lib/chunkio/src/CMakeFiles/chunkio-static.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```
